File: src/main/java/rr/language/inspections/NonExistingBuildingLevel.java
package rr.language.inspections;

import com.intellij.codeInspection.LocalInspectionToolSession;
import com.intellij.codeInspection.ProblemHighlightType;
import com.intellij.codeInspection.ProblemsHolder;
import org.jetbrains.annotations.NotNull;
import rr.language.RRUtil;
import rr.language.psi.RRBuildingLevelRef;
import rr.language.psi.RRBuildingTreeRef;
import rr.language.psi.RRVisitor;

import java.util.Collection;
import java.util.stream.Collectors;

public class NonExistingBuildingLevel extends Inspector {
    @NotNull
    protected RRVisitor buildRRVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
        return new RRVisitor() {
            @Override
            public void visitBuildingLevelRef(@NotNull RRBuildingLevelRef element) {
                Collection<String> all = RRUtil.findAllBuildingLevels(element.getProject()).stream()
                    .map(it -> it.getBuildingLevelNameDecl().getText())
                    .collect(Collectors.toList());

                if (!all.contains(element.getText())) {
                    holder.registerProblem(element, "Non existing building level", ProblemHighlightType.ERROR);
                }
            }
        };
    }
}

File: src/main/java/rr/language/inspections/NonExistingBuildingLevel.kt
package rr.language.inspections

import com.intellij.codeInspection.LocalInspectionToolSession
import com.intellij.codeInspection.ProblemHighlightType
import com.intellij.codeInspection.ProblemsHolder
import rr.language.RRUtil.findAllBuildingLevels
import rr.language.psi.RRBuildingLevel
import rr.language.psi.RRBuildingLevelRef
import rr.language.psi.RRVisitor
import java.util.stream.Collectors

class NonExistingBuildingLevel : Inspector() {
    override fun buildRRVisitor(holder: ProblemsHolder, session: LocalInspectionToolSession): RRVisitor {
        return object : RRVisitor() {
            override fun visitBuildingLevelRef(element: RRBuildingLevelRef) {
                val all: Collection<String> = findAllBuildingLevels(element.project).stream()
                    .map { it: RRBuildingLevel -> it.buildingLevelNameDecl.text }
                    .collect(Collectors.toList())
                if (!all.contains(element.text)) {
                    holder.registerProblem(element, "Non existing building level", ProblemHighlightType.ERROR)
                }
            }
        }
    }
}